Microsoft.Azure.CosmosRepository/src/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -61,7 +63,38 @@ public static class ServiceCollectionExtensions
         }
 
         /// <summary>
-        /// Adds the services required to consume any number of <see cref="IRepository{TItem}"/> 
+        /// Adds the services required to run the in memory implementation of the cosmos repository.
+        /// </summary>
+        /// <param name="services">The service collection to add services to.</param>
+        /// <returns></returns>
+        public static IServiceCollection AddInMemoryCosmosRepository(this IServiceCollection services)
+        {
+            if (services is null)
+            {
+                throw new ArgumentNullException(
+                    nameof(services), "A service collection is required.");
+            }
+
+            services.AddSingleton(typeof(IRepository<>), typeof(InMemoryRepository<>))
+                .AddSingleton<IRepositoryFactory, DefaultRepositoryFactory>();
+
+            return services;
+        }
+
+        /// <summary>
+        /// Remove all of <see cref="IRepository{TItem}"/> from the container.
+        /// </summary>
+        /// <param name="services">The service collection to add services to.</param>
+        /// <returns></returns>
+        public static IServiceCollection RemoveCosmosRepositories(this IServiceCollection services)

```

Comment:
What is this for? I'm not sure I like the idea of exposing this API. If it's intended to be internal for our testing of the in-memory bits, it might be best to keep it internal and expose our internals to our test project.

Reply:
It isn't internal I put it there for convivence it removes the service descriptors for `typeof(IRepository<>)` to enable integration testing code to replace the real ones with the in-memory version. Happy to remove this from the library some people may find it useful.

Reply:
@mumby0168 not clear on what scenarios require a consumer to replace the real ones, versus build it with the in-memory version explicitly. Are we thinking that many will have a "baseline" `IServiceCollection` composed, to which they want to then mutate to swap out the Cosmos impl with the in-memory one during test? Would a compound `ReplaceCosmosRepositoriesWithInMemoryRepositories` method better state this intent?

Reply:
This scenario would only be used for integration tests when they would have already registered the real version they will want to replace it.